addons/mrp_subcontracting_purchase/tests/test_mrp_subcontracting_purchase.py

```diff
@@ -300,6 +300,38 @@ def test_subcontracting_resupply_price_diff(self):
         price_diff_line = invoice.line_ids.filtered(lambda m: m.account_id == stock_price_diff_acc_id)
         self.assertEqual(price_diff_line.credit, 20)
 
+    def test_subcontract_product(self):
+        """ Create a PO for subcontracted product, receive the product (finish MO),
+            create vendor bill, confirm the bill.
+        """
+        product_category_all = self.env.ref('product.product_category_all')
+        product_category_all.property_cost_method = 'average'
+        product_category_all.property_valuation = 'real_time'
+        self.comp1.standard_price = 5
+        purchase = self.env['purchase.order'].create({
+            'partner_id': self.subcontractor_partner1.id,
+            'order_line': [Command.create({
+                'name': self.finished.name,
+                'product_id': self.finished.id,
+                'product_qty': 10,
+                'product_uom': self.finished.uom_id.id,
+                'price_unit': 1,
+            })],
+        })
+        purchase.button_confirm()
+        # receive product
+        receipt = purchase.picking_ids
+        receipt.move_ids.picked = True
+        receipt.button_validate()
+        # create bill
+        purchase.action_create_invoice()
+        aml = self.env['account.move.line'].search([('purchase_line_id', '=', purchase.order_line.id)])
+        aml.move_id.invoice_date = Date.today()
+        aml.move_id.action_post()
+
+        self.assertEqual(self.finished.quantity_svl, 10)
+        self.assertEqual(self.finished.standard_price, 1 + 5)
```

Comment:
The almost exact same test already exists [here](https://github.com/odoo/odoo/blob/17.0/addons/mrp_subcontracting_purchase/tests/test_mrp_subcontracting_purchase.py#L311-L373).
Rather than bloat the tests even more, just update the price of `self.finished` in that test and adapt it to test your fix as well. Both tests would test the exact same workflow otherwise.

Reply:
np :) I've edited the existing test